auto-dark-light@gihaume: v2.0.0

guillaume-mueller opened the pull request:
- Closes #6758
- Closes #7247
- Closes #7923

Note that I used JS minification so the applet loads faster. I understand it looks like compiled binary (that you avoid) but unlike the latter it should be fully reproducible on any system. I added an automatic checksum so anyone can verify the integrity easily.

guillaume-mueller committed:
auto-dark-light@gihaume: v2.0.0
- Closes #6758
- Closes #7247
- Closes #7923

guillaume-mueller committed:
Add PR id in `CHANGELOG.md`

guillaume-mueller committed:
Fix forgotten to copy new timezone location database to runtime files

guillaume-mueller commented:
@rcalixte What do you mean by marking the PR as draft?

rcalixte commented:
> @rcalixte What do you mean by marking the PR as draft?

User-facing code should be readable. The benefits for minification in JavaScript are related to download times not execution times. As the Spices are distributed as compressed files, this isn't a necessary or gainful modification, reproducible or not. Hopefully this makes sense.

guillaume-mueller committed:
Remove minification

guillaume-mueller committed:
Remove slight relicat of Terser use for minification

guillaume-mueller commented:
I wasn't talking about runtime performance but loading time, but anyway it was like 50 ms versus 100 ms, so no problem.

mtwebster commented:
I don't know what QRedshift was doing to cause such an impact, but if it was doing that it was simply very poorly written. You could WTF as much as you want, that's not a native applet. This is one of the drawbacks of supporting third-party applets. We try to watch for bad practices and traps but it's not easy to catch, especially when it happens gradually.

I realize our documentation isn't as great as it could be - unfortunately the best documentation is looking at Cinnamon's core ui code, and its applets and desklets. We try to make getting at system services and states pretty easy, and if it makes sense, we'll make it so it's available for spices to use (like the screensaver or session state). Questions like 'How can I...' brought up in Github discussions I'll usually try to answer also.

I did look at the time change listener, I realize it only triggers for that specific event. I think this is too far to go for efficiency. An extra process running on the remote (or at least, fairly uncommon) chance that the time gets updated. 99% of the time if this triggers, it will be because your timezone changed (which you're also listening separately for). Very likely the timezone monitor is enough.

```
GLib.timeout_add_seconds(GLib.PRIORITY_DEFAULT, 60 * 30, this._periodic_update.bind(this));
....
....
_periodic_update() {
    if (check_the_current_time_against_our_schedule()) {
        /* Do work */
    }

    return GLib.SOURCE_CONTINUE;
}
```
Doing something as simple as comparing times every 30 minutes (or even every 5) will *not* negatively impact your desktop.

guillaume-mueller committed:
Replace C++ passive `Time_change_listener` by JS active `Wall_clock_adjustment_monitor`

guillaume-mueller commented:
Could https://github.com/linuxmint/Cinnamon/blob/master/js/misc/gnomeSession.js really be useful here? I've not found that's the case so I still use a subscription to DBus directly for the sleep entering and returning (waking up) events.

guillaume-mueller committed:
Change `Sleep_events_listener`:

- Change `Screen_lock_checker` class direct use of DBus for `imports.misc.screenSaver` instead and rename it `Screen_lock_change_listener`

- Wrap `Screen_lock_change_listener` class in an awaitable `Screen_unlock_waiter`

- Make `Sleep_events_listener` class only be the DBus wrapper then compose the whole wanted logic in a `Sleep_and_lock_handler`

- Add unit tests in CinnamonJS context, so go back to JS with TS via JSDoc for this

- Add a class diagram

guillaume-mueller committed:
Bump dependencies

guillaume-mueller committed:
Refactor `Keybinding_handler`

guillaume-mueller committed:
Reorganize et rename classes in `src/lib/sys` folder

guillaume-mueller committed:
Improve handling of classes which implements `Disposable`

guillaume-mueller committed:
- Convert some `sys` files back to JS
- Correct command launching error handling and add tests
- Improve `CONTRIBUTING.md`

guillaume-mueller committed:
Replace DBus `login1` `PrepareForSleep` by Cinnamon `misc.gnomeSession` `Presence` proxy

guillaume-mueller committed:
Improve `CONTRIBUTING.md`

guillaume-mueller committed:
Bump dependencies

guillaume-mueller committed:
Revert "Replace DBus `login1` `PrepareForSleep` by Cinnamon `misc.gnomeSession` `Presence` proxy"

This reverts commit e6bbae98920097f2b005759fedaf94c52704ed88.

guillaume-mueller committed:
Clean imports

guillaume-mueller committed:
Bump dependencies

guillaume-mueller committed:
Change folders organization

guillaume-mueller committed:
Change imports to include files extensions in paths

guillaume-mueller committed:
Use TypeScript Import Sorter to improve imports orders

guillaume-mueller committed:
Move main app file

guillaume-mueller committed:
Use JS instead of TS for runtime things

guillaume-mueller committed:
- Fix MobX observables shouldn't be initialized at `undefined`
- Bump dependencies

guillaume-mueller committed:
Remove Vite build

guillaume-mueller committed:
Use ESM imports for GLib and Gio

guillaume-mueller committed:
Update translation files
